src/auth/src/token_cache.rs

```diff
@@ -102,7 +102,15 @@ where
 async fn wait_for_next_token(
     mut rx_token: watch::Receiver<Option<Result<(Token, EntityTag)>>>,
 ) -> Result<(Token, EntityTag)> {
-    rx_token.changed().await.unwrap();
+    if rx_token.changed().await.is_err() {
```

Comment:
Wont gax call wait here still?

Reply:
Yeah change for unwrap is ok. But still the gax call is going to wait here right?